### ISSUE TITLE
[使用 ]Keras 函数式API与其对应的sequence模型训练时区别问题。

### ISSUE THREAD

**Issue by FlyBird-S** [Keras-Theano-TF-DeepLearner/DL-Discuss-Implement-Group#20]
已解决。自己眼拙

**Comment by FlyBird-S** [comment]
```
model = models.Sequential()
model.add(
    layers.Conv2D(32, (3, 3), activation='relu', input_shape=(150, 150, 3)))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(64, (3, 3), activation='relu'))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(128, (3, 3), activation='relu'))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(128, (3, 3), activation='relu'))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Flatten())
model.add(layers.Dense(512, activation='relu'))
model.add(layers.Dense(1, activation='sigmoid'))
```

**Comment by FlyBird-S** [comment]
基于TensorFlow2.0
```
import tensorflow as tf

keras = tf.keras
layers = keras.layers
models = keras.models
optimizers = keras.optimizers
Input = keras.Input
Model = keras.Model
Callbacks = keras.callbacks  # 回调函数
plot_model = keras.utils.plot_model


board_path = 'board'
# model = models.Sequential()
# model.add(
#     layers.Conv2D(32, (3, 3), activation='relu', input_shape=(150, 150, 3)))
# model.add(layers.MaxPooling2D((2, 2)))
# model.add(layers.Conv2D(64, (3, 3), activation='relu'))
# model.add(layers.MaxPooling2D((2, 2)))
# model.add(layers.Conv2D(128, (3, 3), activation='relu'))
# model.add(layers.MaxPooling2D((2, 2)))
# model.add(layers.Conv2D(128, (3, 3), activation='relu'))
# model.add(layers.MaxPooling2D((2, 2)))
# model.add(layers.Flatten())
# model.add(layers.Dense(512, activation='relu'))
# model.add(layers.Dense(1, activation='sigmoid'))

Callbacks_list = [
	#Callbacks.EarlyStopping(monitor='acc',patience=1),
	Callbacks.ModelCheckpoint(filepath='my_model.h5',monitor='val_loss',save_best_only=True),
	Callbacks.TensorBoard(log_dir=board_path,histogram_freq=1)
]
# 网络配置
input_tensor = Input(shape=(150, 150, 3))
x = layers.Conv2D(32, (3, 3), activation='relu')(input_tensor) ## 没有input_shape=(150, 150, 3) 会报错
x = layers.MaxPooling2D((2, 2))(x)
x = layers.Conv2D(64, (3, 3), activation='relu')(x)
x = layers.MaxPooling2D((2, 2))(x)
x = layers.Conv2D(128, (3, 3), activation='relu')(x)
x = layers.MaxPooling2D((2, 2))(x)
x = layers.Conv2D(128, (3, 3), activation='relu')(x)
x = layers.MaxPooling2D((2, 2))(x)

x = layers.Flatten()(x)
x = layers.Dense(512, activation='relu')(x)
out_tensor = layers.Dense(1, activation='softmax')(x)

model = Model(input_tensor, out_tensor)
model.summary()
model.compile(loss='binary_crossentropy',
              optimizer='sgd',
              metrics=['acc'])

# # 绘制网络结构图
plot_model(model,show_shapes=True,to_file='model.png')


# 训练样品处理
ImageDataGenerator = keras.preprocessing.image.ImageDataGenerator

train_datagen = ImageDataGenerator(rescale=1. / 255)
train_generator = train_datagen.flow_from_directory(
    'C:/Users/Song/Desktop/week/keras_test/kaggle/train_dir',
    target_size=(150, 150),
    batch_size=20,
    class_mode='binary')
validation_datagen = ImageDataGenerator(rescale=1. / 255)
validation_generator = train_datagen.flow_from_directory(
    'C:/Users/Song/Desktop/week/keras_test/kaggle/test_dir',
    target_size=(150, 150),
    batch_size=20,
    class_mode='binary')
# import cv2
# for i,y in validation_generator:
#     cv2.imshow('x',i[0])
#     print(y[0])
#     cv2.waitKey()
    
history = model.fit_generator(train_generator,
                              steps_per_epoch=100,
                              epochs=30,
                              validation_data=validation_generator,
                              validation_steps=10,
							  callbacks=Callbacks_list)
model.save('cat_and_dog_2.h5')

```

**Comment by FlyBird-S** [comment]
同样的第一轮结构
使用sequential
loss: 0.6934 - acc: 0.4940 - val_loss: 0.6906 - val_acc: 0.5900
使用函数式API
loss: 7.9963 - acc: 0.4785 - val_loss: 6.5166 - val_acc: 0.5750
